projects/cdk/directives/copy-processor/copy-processor.directive.ts

```diff
@@ -0,0 +1,29 @@
+import {Directive, HostListener, Inject, Input} from '@angular/core';
+import {WINDOW} from '@ng-web-apis/common';
+import {tuiDefaultProp} from '@taiga-ui/cdk/decorators';
+import {TuiStringHandler} from '@taiga-ui/cdk/types';
+import {identity} from 'rxjs';
+
+// @dynamic
+@Directive({
+    selector: '[tuiCopyProcessor]',
+})
+export class TuiCopyProcessorDirective {
+    @Input()
+    @tuiDefaultProp()
+    tuiCopyProcessor: TuiStringHandler<string> = identity;
+
+    constructor(@Inject(WINDOW) private readonly windowRef: Window) {}
+
+    @HostListener('copy.prevent', ['$event'])
+    onCopy(event: ClipboardEvent) {
```

Comment:
Could you create a simple unit test on karma.js?

Reply:
done